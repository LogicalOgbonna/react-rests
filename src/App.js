import React from 'react';
import Header from './component/header/header';
import "./App.scss";
import Headline from './component/headline/headline';
import Button from './component/button/button';
import ListItem from './component/list-item/list-item';
import { connect } from 'react-redux';
import { fetchPost } from './actions';
export class App extends React.Component {
  state = {
    value: []
  }

  onClick = () => {
    this.setState({ value: [{ name: "Man" }] })
  }
  render() {
    return (
      <div className="App">
        <Header />
        <section className="main">
          <Headline header="Posts" description="Click to render Posts" />
          <Button />
          <ListItem />
        </section>

        <button data-test="add-button" onClick={this.onClick}>Click</button>
      </div>
    );

  }
}

const mapStateToProps = state => ({
  posts: state.posts
})

export default connect(mapStateToProps, { fetchPost })(App);
