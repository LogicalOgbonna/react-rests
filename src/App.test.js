import React from "react";
import { App } from "./App";
import { shallow } from 'enzyme';
import { renderWithoutCrashing, findByTestAttr } from "./utils/tests";

describe('Renders Without Crashing', () => {
    let component;
    beforeEach(() => {
        component = shallow(<App />)
    });
    renderWithoutCrashing(component, "<App />");

    it('should initialize `state` with a empty array', () => {
        expect(component.state().value).toEqual([]);
    });

    it('should add a value to `state` array', () => {
        const button = findByTestAttr(component, "add-button");
        button.simulate("click");

        expect(component.state().value.length).toEqual(1);
    })


})
