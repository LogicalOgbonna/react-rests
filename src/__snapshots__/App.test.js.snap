// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Without Crashing should render <App /> Without Crashing 1`] = `undefined`;
